### PR TITLE
Spelling

### PR DESCRIPTION
**Pull request by jsoref**
This PR corrects misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).

The misspellings have been reported at https://github.com/jsoref/barman/commit/ff59f2fafe179dc9b11657ed3cd18c68d8365f47#commitcomment-76465864

The action reports that the changes in this PR would make it happy: https://github.com/jsoref/barman/commit/8688b2936a75c121ff3cb0a27e68f861b394193f

Note: this PR does not include the action. If you're interested in running a spell check on every PR and push, that can be offered separately.

I will be annotating this PR, and I fully expect to be asked to drop certain files.

### CODE REVIEW

File: barman/backup.py [hunk 1]
```diff
@@ -874,7 +874,7 @@ def delete_wal(self, wal_info):
                 os.removedirs(os.path.dirname(wal_info.fullpath(self.server)))
             except OSError:
                 # This is not an error condition
-                # We always try to remove the the trailing directories,
+                # We always try to remove the trailing directories,
```
Comment (jsoref) [inline barman/backup.py:877]: I'm still experimenting with duplicate word detection, but I'm fairly confident about this...

File: barman/cloud_providers/aws_s3.py [hunk 2]
```diff
@@ -140,7 +140,7 @@ def _extra_upload_args(self):
         Return a dict containing ExtraArgs to be passed to certain boto3 calls
 
         Because some boto3 calls accept `ExtraArgs: {}` and others do not, we
-        return a nexted dict which can be expanded with `**` in the boto3 call.
+        return a nested dict which can be expanded with `**` in the boto3 call.
```
Comment (jsoref) [inline barman/cloud_providers/aws_s3.py:143]: I'm less confident about this (it was an automatically suggested item)

File: NEWS [hunk 3]
```diff
@@ -476,29 +476,29 @@ Version 2.4 - 25 May 2018
 
 Relevant bug fixes:
 
--   Fix remote get_file_content method (Github #151), preventing
+-   Fix remote get_file_content method (GitHub #151), preventing
```
Comment (jsoref) [inline NEWS:479]: brand

**Review by jsoref — COMMENTED** [review]
Most corrections were automatically suggested by Google Sheets, but all fault is mine.

File: barman/annotations.py [hunk 4]
```diff
@@ -201,7 +201,7 @@ def get_annotation(self, backup_id, key, use_cache=True):
                 and (backup_id, key) not in self.annotation_cache
             ):
                 return None
-        # We either know there's an annotation or we havn't used the cache so read
+        # We either know there's an annotation or we haven't used the cache so read
```
Comment (jsoref) [inline barman/annotations.py:204]: I noticed this when I first looked at the repository, but Google Sheets didn't suggest fixing it, and somehow I forgot that I really wanted to fix it...

File: barman/command_wrappers.py [hunk 5]
```diff
@@ -714,7 +714,7 @@ class PostgreSQLClient(Command):
     Sometimes the name of a command has been changed during the PostgreSQL
     evolution. I.e. that happened with pg_receivexlog, that has been renamed
     to pg_receivewal. In that case, we should try using pg_receivewal (the
-    newer auternative) and, if that command doesn't exist, we should try
+    newer alternative) and, if that command doesn't exist, we should try
```
Comment (jsoref) [inline barman/command_wrappers.py:717]: Probably?

File: tests/test_copy_controller.py [hunk 6]
```diff
@@ -544,7 +544,7 @@ def test_fill_buckets(self):
         file_list = []
         total_size = 0
         for i in range(1001):
-            # We are using a prime number to get a non-correlable distribution
+            # We are using a prime number to get a non-correlatable distribution
```
Comment (jsoref) [inline tests/test_copy_controller.py:547]: Google Suggest suggests it (sometimes I'm disappointed that Google Sheets and Google Suggest don't use the same set).

**Review by jsoref — COMMENTED** [review]
The additional fixes were slightly more manual (they were identified by the same tool, but the spreadsheet I used to get corrections didn't offer suggestions, and I neglected to manually look over the list of possible tokens...)

**Comment by mikewallace1979** [comment]
@jsoref Thanks for making this PR - I haven't looked in detail but the majority of these look like valid corrections. We're focused on getting 3.0.0 out the door this week but once the dust has cleared after that we'll take a look at the suggestions here and hopefully get this merged.

File: barman/cloud_providers/aws_s3.py (hunk 2 above)
Comment (mikewallace1979) [reply barman/cloud_providers/aws_s3.py:143]: I think nested is the right thing here :+1:

**Review by mikewallace1979 — APPROVED** [review]
Thanks for this PR @jsoref - I don't see anything here we wouldn't want to merge.

**Pull request closed without merging**